extensions/iceberg/build.gradle

```diff
@@ -58,6 +58,31 @@ dependencies {
     testImplementation project(':engine-test-utils')
 }
 
+/**
+ * We are setting up a "security-manage-allow" tag to add the system property `-Djava.security.manager=allow`
+ * to the tests that need it for Java 23+. Otherwise, the following exception is thrown:
+ *
+ * java.lang.UnsupportedOperationException: getSubject is supported only if a security manager is allowed
+ *   at java.base/javax.security.auth.Subject.getSubject(Subject.java:347)
+ *   at org.apache.hadoop.security.UserGroupInformation.getCurrentUser(UserGroupInformation.java:588)
+ *
+ * See https://issues.apache.org/jira/browse/HADOOP-19212
+ * See https://jdk.java.net/23/release-notes#JDK-8296244
```

Comment:
Note that this is going away entirely "soon", so this is a stopgap that we'll need to solve differently in the nearish future https://openjdk.org/jeps/486

Reply:
Yes; ultimately, the underlying libraries in use (hadoop in this case) will need to be updated to not use security manager.